modules/analysis/src/endpoints/test.rs

```diff
@@ -677,5 +677,33 @@ async fn spdx_package_of(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
 
     assert_eq!(1, matches.len());
 
+    let uri = format!(
+        "/api/v2/analysis/root-component?q={}",
+        urlencoding::encode("SATELLITE-6.15-RHEL-8")
+    );
+    let request: Request = TestRequest::get().uri(&uri).to_request();
+    let response: Value = app.call_and_read_body_json(request).await;
+    log::info!("{}", serde_json::to_string_pretty(&response)?);
+
+    let sbom = &response["items"][0];
+    let matches: Vec<_> = sbom["ancestors"]
+        .as_array()
+        .into_iter()
+        .flatten()
+        .filter(|m| {
+            m == &&json!({
+              "sbom_id":  sbom["sbom_id"],
+              "node_id": m["node_id"],
+              "relationship": "PackageOf",
+              "purl": m["purl"], // long list assume it's correct
+              "cpe": m["cpe"], // long list assume it's correct
+              "name": "rubygem-google-cloud-compute",
+              "version": "0.5.0-1.el8sat"
+            })
```

Comment:
To me, this is a brittle test. I'd want my expectation to be the minimum required to affirm the test. If some future change adds or takes away one of the fields that has nothing to do with this test, it's still gonna break. Is that good or annoying? I'd rather something like this, maybe even omitting `name` and `version`, too:
```suggestion
            m["sbom_id"] == sbom["sbom_id"]
                && m["relationship"] == "PackageOf"
                && m["name"] == "rubygem-google-cloud-compute"
                && m["version"] == "0.5.0-1.el8sat"
```

Reply:
I kinda was thinking about similar lines.  Is there an existing function that can test if an actual Value matches a partial set of fields of another expected Value?  The assertion would become more concise and less brittle then.

Reply:
Not that I know of. You can always downcast the `Value` to an `AncestorSummary` or `DepSummary`, but I'm not sure `m["name"]` is all that better or worse than `m.name`.

Another option is to use [`Value::pointer`](https://docs.rs/serde_json/latest/serde_json/enum.Value.html#method.pointer), but I haven't personally tried that.